packages/terra-signature/src/Signature.jsx

```diff
@@ -0,0 +1,204 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import 'terra-base/lib/baseStyles';
+import './Signature.scss';
+
+const LineWidths = {
+  EXTRAFINE: 1,
+  FINE: 2,
+  MEDIUM: 4,
+  HEAVY: 6,
+};
+
+
+const propTypes = {
+  /**
+   * The line width to use when drawing the signature on the canvas.
+   * One of LineWidths.EXTRAFINE, LineWidths.FINE, LineWidths.MEDIUM, LineWidths.HEAVY.
+   */
+  lineWidth: PropTypes.oneOf([LineWidths.EXTRAFINE, LineWidths.FINE, LineWidths.MEDIUM, LineWidths.HEAVY]),
+
+  /**
+  * Line segments that define signature.
+  */
+  lineSegments: PropTypes.array,
+
+  /**
+   * A callback function to execute when a line segment is drawn. The first parameter is the event, the
+   * second parameter is all the line segments, and the last parameter is the most recent line segment drawn.
+   */
+  onChange: PropTypes.func,
+
+};
+
+const defaultProps = {
+  lineWidth: LineWidths.FINE,
+  lineSegments: undefined,
+  onChange: undefined,
+};
+
+class Signature extends React.Component {
+
+  constructor() {
+    super();
+
+    this.state = { lineSegments: [] };
+  }
+
+  componentDidMount() {
+    const canvas = this.node;
+
+    this.node.addEventListener('mousedown', e => this.mouseDown(e));
+    this.node.addEventListener('mousemove', e => this.mouseXY(e));
+    document.body.addEventListener('mouseup', e => this.mouseUp(e));
+
+    // For mobile
+    this.node.addEventListener('touchstart', e => this.mouseDown(e), false);
+    this.node.addEventListener('touchmove', e => this.mouseXY(e), true);
+    this.node.addEventListener('touchend', e => this.mouseUp(e), false);
+    document.body.addEventListener('touchcancel', e => this.mouseUp(e).bind(this), false);
+
+    this.node.width = canvas.getBoundingClientRect().width;
+    this.node.height = canvas.getBoundingClientRect().height;
+
+    const context = this.node.getContext('2d');
+
+    context.lineWidth = this.props.lineWidth;
+
+    const style = window.getComputedStyle(this.node);
+    const color = style.getPropertyValue('color');
+    context.strokeStyle = color;
+
+    if (this.props.lineSegments) {
+      this.state.lineSegments = this.props.lineSegments;
+      this.drawSignature(this.state.lineSegments);
+    }
+
+    window.addEventListener('resize', () => this.updateDimensions());
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.lineSegments !== nextProps.lineSegments) {
```

Comment:
You may need to compare the values and length, rather than compare references.

Reply:
Recording a signature with line segments takes a lot of line segments!  We were a bit worried about the size of the array but with a variety of real world examples, it typically results in a 50kb or less array.  Still a concern that during the course of signing, there might be several arrays hanging around in the browser memory but we plan to test this aggressively when we add the terra-signature component to esignature.  

But more to your original point, whether we need to compare the actual element values in the array.  Do you know how expensive array comparison is in javascript or should we look for some simple comparisons like length to decide if we are dealing with the same signature?  Thoughts?

Reply:
I'd say maybe start with at least comparing the length, but if we are worried about array performance we may need to consider switching to an updated key/forcing a remount on clear. As this comparison will occur with every state change and render.

Reply:
I will add a length comparison in addition to reference.  Given the vast number of line segments to describe a signature, it turns out to be a fairly good evaluator for sameness. 

I will revisit when the signature component is added to the eSignature app.